src/classes/BDI_Donations.cls

```diff
@@ -479,32 +467,28 @@ public class BDI_Donations {
     */
     private void setDryRunDonationStatuses(List<DataImport__c> dataImportDonations) {
 
+        Map<String, String> importStatusToDryRunStatus = new Map<String, String>{
```

Comment:
You could move this outside the method to avoid reconstructing it each time the method is called.

Reply:
@lmeerkatz I did try this last time, and had some really surprising test failures! I'm not sure why, but it seems to be happiest in this method 😬